text/0043-block-metadata.md

```diff
@@ -0,0 +1,193 @@
+key proves that the block and metadata were generated by the holder of that
+private key.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When consensus nodes externalize a block, they will include `BlockMetadata` with
+the AVR and quorum set for that block.
+
+LedgerDB and WatcherDB also track and persist the block metadata.
+
+Note that while the `BlockMetadata` can be used to verify chain integrity, it
+couldn't be used to falsify any new transactions or blocks.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Add a `message BlockMetadata` to `blockchain.ArchiveBlock`, and an analogous
+`struct BlockMetadata` to `mc_blockchain_types::BlockData`.
+
+## Protobuf schema
+
+```protobuf
+message BlockMetadataContents {
+    /// The Block ID.
+    BlockID id = 1;
+
+    /// Quorum set configuration at the time of externalization.
+    QuorumSet quorum_set = 2;
+
+    /// IAS report for the enclave which generated the signature.
+    VerificationReport verification_report = 3;
+}
+
+message BlockMetadata {
+    /// Metadata signed by the consensus node.
+    BlockMetadataContents contents = 1;
+
+    /// Message signing key.
+    Ed25519Public node_key = 2;
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    Ed25519Signature signature = 3;
+}
+
+// NB: Only addition is field #4.
+message ArchiveBlockV1 {
+    /// Block.
+    Block block = 1;
+
+    /// Contents of the block.
+    BlockContents block_contents = 2;
+
+    /// Block signature.
+    BlockSignature signature = 3;
+
+    /// Additional signed metadata about this block.
+    BlockMetadata metadata = 4;
+}
+```
+
+Since this is a protobuf-additive change, we don't need a new `ArchiveBlock`
+version/type.
+
+## Signing metadata
+The metadata signature will always differ from the block signature. The block
+signature is generated by the enclave using a private key that never leaves the
+enclave, while the metadata is signed by the "untrusted" code outside the
+enclave using the SCP message signing key, since the enclave does not have this
+metadata.
+
+The metadata signature is generated from the `Digestible` encoding of
+`BlockMetadataContents`, since the protobuf wire format is not a canonical
+encoding.
+
+## Signing key history
+MobileCoin Foundation will publish a mapping/list of node_id to a range of block
+indexes, which node operators and consumers can use to verify the metadata
+signatures.
+
+This mapping will contain multiple node_id:block_range entries, in the following
+TOML format:
+```toml
+[[node]]
+message_signing_pub_key = ""  # Pub key, as PEM-encoded data or a file path containing such data.
+first_block_index = X  # Index of the first block signed with this key.
+last_block_index = Y  # Optional index of the last block signed with this key.
+```
+
+## Historical AVRs
+With `BlockMetadata`, consumers of the blockchain no longer need a Watcher DB.
```

Comment:
Its worth noting that the watcher is currently used to provide block timestamps, so I think consumers of the blockchain that need that information will still need it. The root of the issue is that timestamps are optional, and only available for blocks signed by an enclave. When syncing blocks from S3 (e.g. when bootstrapping a new node, or when a node does catchup) I believe this results in it not having timestamps for those blocks.
The current workaround is having the watcher sync blocks from all nodes, and that guarantees that once all blocks have been synced from all nodes for a given block index, at least one of them will have a proper signature/timestamp.

Reply:
Tracing the code - I believe the timestamp gets added within the logic in this call: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/verification_reports_collector.rs#L92 - would that be correct?

And then it checks the signer key every poll interval: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/verification_reports_collector.rs#L220

At the moment we're working on a bootstrap file for this to provide the timestamps historically, but any new client that "replaces watcher" would have to include this attest() call to add the timestamp and explicitly check for AVRs I think. 

So it seems like we'll maybe have to wait until we have a block streaming component actively collecting these AVRs.

Reply:
Yeah, I've arrived at a similar conclusion. I've updated the wording to "consumers of the blockchain may not need a Watcher DB."